### ISSUE TITLE
Assertion failure in ITE term simplification

### ISSUE THREAD

**Issue by dddejan** [SRI-CSL/yices2#61]
There seems to be an issue with what ITE simplification in the term manager expects, and what MCSAT creates in pre-processing. 

Problem file https://gist.github.com/dddejan/627f9b17a9c154b9c6a93836e495066f

Yices fails with 
```
yices_smt2: terms/term_manager.c:1741: mk_integer_polynomial_ite: Assertion `is_integer_term(tbl, t) && is_integer_term(tbl, e)' failed.
```
Backtrace:
```
#0  0x00007ffff7541428 in __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:54
#1  0x00007ffff754302a in __GI_abort () at abort.c:89
#2  0x00007ffff7539bd7 in __assert_fail_base (fmt=<optimized out>, assertion=assertion@entry=0x62ba30 "is_integer_term(tbl, t) && is_integer_term(tbl, e)", 
    file=file@entry=0x62b5c9 "terms/term_manager.c", line=line@entry=1741, function=function@entry=0x62c880 <__PRETTY_FUNCTION__.10749> "mk_integer_polynomial_ite") at assert.c:92
#3  0x00007ffff7539c82 in __GI___assert_fail (assertion=0x62ba30 "is_integer_term(tbl, t) && is_integer_term(tbl, e)", file=0x62b5c9 "terms/term_manager.c", line=1741, 
    function=0x62c880 <__PRETTY_FUNCTION__.10749> "mk_integer_polynomial_ite") at assert.c:101
#4  0x000000000051a4b9 in mk_integer_polynomial_ite (manager=0x90bf68, c=824, t=820, e=818) at terms/term_manager.c:1741
#5  0x000000000051b023 in mk_ite (manager=0x90bf68, c=824, t=820, e=818, tau=1) at terms/term_manager.c:2046
#6  0x0000000000574abf in mk_composite (pre=0x90bf60, kind=ITE_TERM, n=3, children=0x941440) at mcsat/preprocessor.c:107
#7  0x0000000000575359 in preprocessor_apply (pre=0x90bf60, t=740, out=0x7fffffffda70) at mcsat/preprocessor.c:318
#8  0x000000000055219e in mcsat_assert_formulas (mcsat=0x90bdd0, n=1, f=0x8fd710) at mcsat/solver.c:1667
#9  0x0000000000429786 in context_process_assertions (ctx=0x90a9d0, n=1, a=0x8fd710) at context/context.c:5597
#10 0x0000000000429c39 in assert_formulas (ctx=0x90a9d0, n=1, f=0x8fd710) at context/context.c:5806
#11 0x0000000000413b8a in yices_assert_formulas (ctx=0x90a9d0, n=1, t=0x8fd710) at api/yices_api.c:7308
#12 0x000000000057c323 in check_delayed_assertions (g=0x8e32e0 <__smt2_globals>) at frontend/smt2/smt2_commands.c:2687
#13 0x000000000057ffa0 in smt2_check_sat () at frontend/smt2/smt2_commands.c:4887
#14 0x00000000005876de in eval_smt2_check_sat (stack=0x8e1fc0 <stack>, f=0x90d800, n=0) at frontend/smt2/smt2_term_stack.c:1023
#15 0x0000000000473dac in tstack_eval (stack=0x8e1fc0 <stack>) at parser_utils/term_stack2.c:5351
#16 0x00000000005834e9 in smt2_parse (parser=0x8e1fa0 <parser>, start=c0) at frontend/smt2/smt2_parser.c:270
#17 0x0000000000584448 in parse_smt2_command (parser=0x8e1fa0 <parser>) at frontend/smt2/smt2_parser.c:839
#18 0x000000000040630c in main (argc=2, argv=0x7fffffffdda8) at frontend/yices_smt2.c:468
```

**Comment by dddejan** [comment]
Issue is when creating an ITE with children
```
0: (>= (+ 19 v0 (* 2 v2)) 0)
1: (+ -1 (ite (p0 v0 t!770) 1 0))
2: (* -2 v2)
```
Child 1 seems to be integer, while child 2 is real.

**Comment by dddejan** [comment]
Fixed by @BrunoDutertre in 2e8048fce0dc381ecddb5445a29a3c59dfd6a1da
